explainer.md

```diff
@@ -1,751 +1,423 @@
 
-#### Offering Page-B elements
+# Compatibility with frameworks
 
-Page-B elements are offered for the transition using the same mechanism as in
-Page-A, `page-transition-tag`.
+```js
+const transition = new SameDocumentTransition();
+await transition.prepare(async () => {
+  await updateTheDOMSomehow();
+});
+```
 
-Elements from Page-A and Page-B that have the same tag are merged into one
-transition elements as documented in "Mixing in elements from Page-B and
-associating them with transition elements from Page-A".
+This pattern assumes the developer is in charge of DOM updates, but that isn't the case with most web frameworks. However, the [demo site featured in this explainer](https://http203-playlist.netlify.app/) was built using Preact, and uses a [React-style hook](https://github.com/jakearchibald/http203-playlist/blob/main/src/client/utils.ts#L11) to wrap the above API and make it usable with React/Preact.
 
-#### Targeting transition elements
+As long as the framework provides a notification when the DOM is updated, which they already do to allow custom handling of elements, the transition API can be made to work with the framework.
 
-The pseudo-elements are constructed as documented in "how are transition
-elements represented".
+# Animating with JavaScript
 
-These pseudo-element selectors provide access to these pseudo-elements via
-`:root`:
+The promise returned by `prepare()` fulfills when both states have been captures and the pseudo-element tree has been successfully built. This provides developers with a point where they can animate those pseudo-elements with the [Web Animation API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API).
 
-- `::page-transition` - The parent of all transition containers.
-- `::page-transition-container(tag)` - Select the 'transition element' of a
-  given `page-transition-tag`.
-- `::page-transition-image-wrapper(tag)` - Select the 'image wrapper'.
-- `::page-transition-incoming-image(tag)` - Select the incoming image.
-- `::page-transition-outgoing-image(tag)` - Select the outgoing image.
+https://user-images.githubusercontent.com/93594/184120371-678f58b3-d1f9-465b-978f-ee5eab73d120.mp4
 
-Using '\*' instead of a tag selects the equivalent element for every tag.
+This example animates a circular `clip-path` from the position of the click to a size that covers the viewport. Since this isn't possible with CSS alone ([although it may get easier in future](https://github.com/w3c/csswg-drafts/issues/824#issuecomment-1204467456)), the Web Animation API is used.
 
-Issue: It would be nice to be consistent with the '\*' CSS selector in that we
-would like `::page-transition(*)` to have a lower specificity than
-`::page-transition(tag)`. 
+# Customizing the transition based on the type of navigation
```

Comment:
Is there a sample page to point to for a code example? Perhaps something that builds on top of the Navigation API and shows how the 2 integrate nicely.

Reply:
Wellllll they don't really integrate nicely right now, as you get that "did the navigation work or not??" issue that we're going to fix in the next iteration of the API.

Reply:
Fair. So we can update this once that issue is fixed in the spec.